lsp/server/src/validateGraphql.ts

```diff
@@ -0,0 +1,110 @@
+
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import {parse, ASTNode} from 'graphql';
+import { gqlPluckFromCodeStringSync } from '@graphql-tools/graphql-tag-pluck';
+import { Diagnostic } from 'vscode-languageserver/node';
+import { DiagnosticProducer } from './diagnostic/DiagnosticProducer';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { MisspelledUiapi } from './diagnostic/gql/misspelled-uiapi';
+
+const diagnosticProducers: DiagnosticProducer<ASTNode>[] = [];
+diagnosticProducers.push(new MisspelledUiapi());
+
+/**
+ * validate the graphql queries in the document.
+ * @param textDocument 
+ * @param maxCount  the max count of diagnostics to return 
+ */
+export async function validateGraphql(
+    textDocument: TextDocument, 
+    maxCount: number
+): Promise<Diagnostic[]> {
+    const results: Diagnostic[] = [];
+
+    if (maxCount <= 0 || diagnosticProducers.length === 0) {
+        return results;
+    }
+
+    // Find the gql``s in the file content
+    const graphQueries = gqlPluckFromCodeStringSync(
+        textDocument.uri,
+        textDocument.getText(),
+        {
+            skipIndent: true
+        }
+    );
+
+    // Validate each query
+    for (const query of graphQueries) {
+        if (results.length >= maxCount) {
+            break;
+        }
+        const lineOffset = query.locationOffset.line - 1;
+        const columnOffset = query.locationOffset.column + 1;
```

Comment:
Why is column offset +1?

Reply:
graphql-eslint has same code.  seems that 

- query.locationOffset's line and column is 1 based. 
- query.locationOffset.column value starts at the `. so need to plus 1 to get the correct body start position.